packages/amplify_datastore/example/lib/models/Blog.dart

```diff
@@ -23,8 +25,16 @@ import 'package:flutter/foundation.dart';
 class Blog extends Model {
   static const classType = const BlogType();
   final String id;
-  final String name;
-  final List<Post> posts;
+  final String? _name;
+  final List<Post>? _posts;
+
+  String get name {
+    return _name!;
```

Comment:
Yes that's correct we can't guarantee it's actually non null.

Reply:
Will that be a problem if this is how codegen'd models look? Unfortunately, I haven't been able to keep up with the codegen changes and I just wanted to double check we're not generating models that have potential to cause app crashing.

Reply:
Yes we agreed that it's okay for now.  

When you query all of its fields will be returned.  Required fields must be set because you couldn't have saved a model without those required fields.  Thus when we force cast the underlying value should not be null. 

However, nested models currently have only their id field returned.  This means it others required fields can be null. 

Our resolution so far is to release this as is and to work on updating the iOS and Android bridge code to actually send back the fields of the nested models.  

Currently iOS does this but Android does not so we are coordinating with Android team to get that done.

Reply:
Ok gotcha. Sometimes I just need to put the conversation in context of code.